icicle/src/poseidon2/constants.cu

```diff
@@ -0,0 +1,120 @@
+#include "poseidon2/poseidon2.cuh"
+
+/// These are pre-calculated constants for different curves
+#include "fields/id.h"
+#if FIELD_ID == BN254
+#include "poseidon2/constants/bn254_poseidon2.h"
+using namespace poseidon2_constants_bn254;
+#elif FIELD_ID == BLS12_381
+#include "poseidon2/constants/bls12_381_poseidon2.h"
+using namespace poseidon2_constants_bls12_381;
+#elif FIELD_ID == BLS12_377
+#include "poseidon2/constants/bls12_377_poseidon2.h"
+using namespace poseidon2_constants_bls12_377;
+#elif FIELD_ID == BW6_761
+#include "poseidon2/constants/bw6_761_poseidon2.h"
+using namespace poseidon2_constants_bw6_761;
+#elif FIELD_ID == GRUMPKIN
+#include "poseidon2/constants/grumpkin_poseidon2.h"
+using namespace poseidon2_constants_grumpkin;
+#elif FIELD_ID == BABY_BEAR
+#include "poseidon2/constants/babybear_poseidon2.h"
+using namespace poseidon2_constants_babybear;
+#endif
+
+namespace poseidon2 {
+  template <typename S>
+  cudaError_t create_optimized_poseidon2_constants(
```

Comment:
is there a way to release the constants that you allocate here? what is the intention here in terms of memory management?

Reply:
no, currently there is no way to release the constants. The intention was to store all the constants on-device like we do with twiddles

Reply:
@yshekel good point here, we need to let users to control the free memory on the device, to be solved in next update